client/components/Modal.jsx

```diff
@@ -0,0 +1,67 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+// gray background
+const backdropStyle = {
+  position: 'fixed',
+  top: 0,
+  bottom: 0,
+  left: 0,
+  right: 0,
+  backgroundColor: 'rgba(0,0,0,0.9)',
+  padding: 50
+};
+
+const modalStyle = {
+  backgroundColor: '#fff',
+  borderRadius: 5,
+  maxWidth: 500,
+  minHeight: 300,
+  margin: '0 auto',
+  backgroundColor: 'rgba(0,0,0,0.5)',
+  padding: 30,
+  position: 'relative'
+};
+
+const footerStyle = {
+  position: 'absolute',
+  bottom: 20
+};
+
+export default class Modal extends React.Component {
+  onClose = e => {
```

Comment:
I'm surprised that this kind of function declaration works inside of an es6 class, and I'd like to learn how it works. Could you show me it  when you have a moment?